crates/walrus-service/src/node/consistency_check.rs

```diff
@@ -200,6 +200,21 @@ fn compose_blob_list_digest_and_check_sliver_data_existence(
             Ok(blob_info) => {
                 hasher.write(blob_info.0.as_ref());
 
+                // Check if the committee onchain has moved forward. If so, we cannot perform the
+                // sliver data existence check because the sliver data may not be fully stored
+                // in `epoch` based on the blobs' certified status in committee epoch.
+                if enable_sliver_data_existence_check && node.current_committee_epoch() > epoch {
+                    enable_sliver_data_existence_check = false;
+                    total_synced_scanned = 0;
+                    total_fully_stored = 0;
+                    tracing::info!(
+                        event_epoch = epoch,
+                        committee_epoch = node.current_committee_epoch(),
+                        "current committee epoch is not the same as the event epoch, skipping \
+                        sliver data existence check"
+                    );
+                }
```

Comment:
Fix 2: what could happen is that during data existence check, the epoch may move to the next epoch, which makes is_blob_certified to return based on next epoch status. In this case, it's possible that certified blob that will be expired may not stored locally, and therefore we skip sliver data existence check.